lib/modrain/index.js

```diff
@@ -0,0 +1,132 @@
+var Rx = require('rx');
+var modrain = {};
+var _ = require('lodash');
+var Promise = require('bluebird');
+
+/**
+ * Represents single node module
+ */
+function ModRain(name, handlers) {
+  this.name = name;
+  // Methods that are mounted
+  this.handlers = handlers ? handlers : {};
+}
+
+/**
+ *
+ * Creating an observer for a single handler.
+ *
+ * @param namespace
+ * @param handlerName
+ * @param originalHandler
+ * @returns {Observable<T>|Rx.Observable<T>}
+ */
+function createObserver(namespace, handlerName, originalHandler) {
+
+  var ns = _.get(modrain, namespace);
+
+  var observable = Rx.Observable.create(function(observer) {
+    //Here, we are wrapping the original handler function in a function that wil call
+    //the original function and also forward the result / error to the observable
+    ns[handlerName] = function() {
+
+      //Applying the original handler with the same context.
+      return originalHandler.apply(this, arguments).then(function(handlerResult) {
+        //Emitting the result to the observer. Any observer subscribers will then also emit.
+        observer.onNext(handlerResult);
+        return Promise.resolve(handlerResult);
+      });
+    };
+
+    return function () {
+      console.log("Disposed", this);
+      //The observer is being disposed. Clean up the observer and re-assign the original function
+      ns._observers[handlerName] = undefined;
+      ns[handlerName] = originalHandler;
+    }
+  }).share();
+
+  return observable;
+}
+
+/**
+ *
+ * Mounting an observe function to observe a handler.
+ *
+ * This allows users to observe a single handler
+ *
+ * E.g. for a namespace called "users" and a handler called "create", the user can write
+ *
+ * modrain.users.observe('create');
+ *
+ * to return a standard Observable.
+ *
+ * @param namespace
+ * @private
+ */
+ModRain.prototype._mountObserve = function(namespace) {
+
+  var ns = _.get(modrain, namespace);
+
+  //Assigning the 'observe' function to the namespace.
+  ns.observe = function(handlerName) {
+    //Here we want to observe a particular handler
+
+    //If the observer already exists, then use that one
+    if (ns._observers[handlerName]) {
+      console.log("Already Exists ", handlerName, ns._observers[handlerName]);
```

Comment:
Is that statement possible to reach considering that line 123 will throw error. 
We should be consistent around duplicate handlers. 
I have no strong opinions about it but I was thinking that having error will prevent application developers from having 2 modules mounting on the same namespace that can lead to unexpected results.